veps/sig-storage/utility-volumes.md

```diff
@@ -0,0 +1,256 @@
+      * `UtilityVolumes` will not have a corresponding entry in the `domain.devices.disks` or similar sections, explicitly indicating they are not for the VM domain.
+
+2.  **Volume Hot-Attachment/Detachment Mechanism:**
+
+      * **Hotplug Process:** `utilityVolumes` use KubeVirt's existing volume hotplug infrastructure. When added to the VM or VMI specification, they are automatically recognized as hotplug volumes and mounted as directories in the `virt-launcher` Pod.
+      * **Mounting Behavior:** Unlike regular volumes that are mounted as filesystems containing a `disk.img` file served to the VM domain, `utilityVolumes` are mounted as filesystem directories directly accessible within the `virt-launcher` Pod environment. This allows data to be written directly to and read from these mounted directories without VM guest involvement.
+      * **Management Options:** 
+          * **VM-level:** Add to `spec.template.spec.utilityVolumes` for persistent volumes that survive VM restarts
+          * **VMI-level:** Add to `spec.utilityVolumes` for temporary, on-demand operations. This action requires a controller with the necessary permissions.
+      * **Lifecycle:** The `virt-controller` watches for specification changes and the `virt-handler` manages Pod volume mounting/unmounting accordingly.
+
+3.  **Status Reporting:**
+
+Utility volumes will be reported in the existing VMI `status.volumeStatus` field alongside regular volumes, using the current `VolumeStatus` struct without modifications.
+      * **Target Field Handling:** For utility volumes, the `Target` field will be empty since these volumes are not presented to the VM guest.
+      * **Standard Status Fields:** Utility volumes will use the existing status fields (`Phase`, `Reason`, `Message`, etc.) to report their lifecycle state.
+
+
+### Migration Compatibility
+
+**Design Principle:** Utility volumes are temporary operational tools attached via pod-level attachment pods and are incompatible with libvirt's migration mechanisms. **Consequently, these volumes will not migrate with the VM.**
+
+**Migration Process:**
+1. **Pre-Migration:** In the unlikely event that a utility operation is active at migration time, each utility type must be designed to handle this possibility, either aborting the operation or ensuring its completion before migration proceeds.
+2. **During Migration:** Utility volumes remain attached to source VM. Migration controller filters utility volumes preventing the creation of attachment pods on destination node, namespace or cluster.
+3. **Migration Completion:** Just before migration transitions to succeeded state, utility volumes are automatically removed from both VMI and VM specifications to prevent post-migration re-attachment.
```

Comment:
This would lead to the resource being out of sync with gitops. Ideally we don't touch any specs. Can we wait for migration to complete before starting hotplug pods for utility volumes?

Reply:
In general, we shouldn't update resource specs unless it's something created/managed internally (not in git). If a user defines a utility volume in git that is later removed by a controller, the resource will become "out of sync." Users can define policies in systems like Argo for what to do in this case. But we should avoid if possible

Reply:
So should we prevent the option of adding utilityvolumes? it should be added only by some supported command via our controllers like for current memorydump and future backup?

Reply:
yeah no utility volumes in VM specs for now is what I propose